onboard-server/src/main/java/no/vegvesen/ixn/serviceprovider/OnboardRestController.java

```diff
@@ -97,11 +93,7 @@ public void deleteCapability(@PathVariable String serviceProviderName, @PathVari
 
 
 		// Updating the Service Provider capabilities based on the incoming capabilities that will be deleted.
-		ServiceProvider serviceProviderToUpdate = serviceProviderRepository.findByName(serviceProviderName);
-		if (serviceProviderToUpdate == null) {
-			logger.warn("The Service Provider trying to delete a capability is new. No capabilities to delete. Rejecting...");
-			throw new NotFoundException("The Service Provider trying to delete a capability does not exist in the database. Rejecting...");
-		}
+		ServiceProvider serviceProviderToUpdate = getOrCreateServiceProvider(serviceProviderName);
```

Comment:
Same here, when trying to delete a Capability

Reply:
This will fail in removeDataType(), which will throw an NotFoundException due to the requested capability not being found in the collection. It's a bit hard to see directly from the code in the provider, I know.